From 3007f114c77e3991dfcaa6ad50c8d6b6cae2dd5c Mon Sep 17 00:00:00 2001
From: Naresh Solanki <naresh.solanki@9elements.com>
Date: Thu, 19 Mar 2026 14:03:56 +0000
Subject: [PATCH 4/9] ibm: Move SPYRE to separate JSON

In order to load once instance of the Spyre card for every VPD EEPROM
move it to a separate file. Thus Bus numbers are now read from Dbus.

Since the Spyre cards contain I2C muxes this change depends on:
Depends-Change-Id: I8a93735ac2f36930d2b9a7ff9342b50e760a85c5
"Support MuxChannel object for dynamic buses"

TEST=Works on IBM system1 for SPYRE1.

Upstream-Status: Pending
Change-Id: I86e3aafa9c2de25e0ad80557829a75ebc9f80eb0
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>


---
 configurations/meson.build            |   1 +
 configurations/system1_baseboard.json | 224 --------------------------
 configurations/system1_spyre.json     |  66 ++++++++
 3 files changed, 67 insertions(+), 224 deletions(-)
 create mode 100644 configurations/system1_spyre.json

diff --git a/configurations/meson.build b/configurations/meson.build
index 4e769c0ca..bc045dabe 100644
--- a/configurations/meson.build
+++ b/configurations/meson.build
@@ -191,6 +191,7 @@ configs = [
     'system1_baseboard.json',
     'system1_chassis.json',
     'system1_psu.json',
+    'system1_spyre.json',
     'tola.json',
     'tyan/tyan_s5549_baseboard.json',
     'tyan/tyan_s7106_baseboard.json',
diff --git a/configurations/system1_baseboard.json b/configurations/system1_baseboard.json
index 7fb2b23ad..40088a9fe 100644
--- a/configurations/system1_baseboard.json
+++ b/configurations/system1_baseboard.json
@@ -552,230 +552,6 @@
                 "CECIO"
             ]
         },
-        {
-            "Address": "0x4b",
-            "Bus": 72,
-            "Name": "SPYRE9_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 73,
-            "Name": "SPYRE9_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 74,
-            "Name": "SPYRE10_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 75,
-            "Name": "SPYRE10_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 76,
-            "Name": "SPYRE11_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 77,
-            "Name": "SPYRE11_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 78,
-            "Name": "SPYRE12_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 79,
-            "Name": "SPYRE12_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 80,
-            "Name": "SPYRE13_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 81,
-            "Name": "SPYRE13_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 82,
-            "Name": "SPYRE14_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 83,
-            "Name": "SPYRE14_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 84,
-            "Name": "SPYRE15_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 85,
-            "Name": "SPYRE15_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 86,
-            "Name": "SPYRE16_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 87,
-            "Name": "SPYRE16_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 88,
-            "Name": "SPYRE1_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 89,
-            "Name": "SPYRE1_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 90,
-            "Name": "SPYRE2_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 91,
-            "Name": "SPYRE2_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 92,
-            "Name": "SPYRE3_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 93,
-            "Name": "SPYRE3_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 94,
-            "Name": "SPYRE4_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 95,
-            "Name": "SPYRE4_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 96,
-            "Name": "SPYRE5_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 97,
-            "Name": "SPYRE5_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 98,
-            "Name": "SPYRE6_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 99,
-            "Name": "SPYRE6_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 100,
-            "Name": "SPYRE7_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 101,
-            "Name": "SPYRE7_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x4b",
-            "Bus": 102,
-            "Name": "SPYRE8_TEMP1",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
-        {
-            "Address": "0x48",
-            "Bus": 103,
-            "Name": "SPYRE8_TEMP2",
-            "PowerState": "On",
-            "Type": "TMP112"
-        },
         {
             "Address": "0x4c",
             "Bus": 70,
diff --git a/configurations/system1_spyre.json b/configurations/system1_spyre.json
new file mode 100644
index 000000000..1d86be6db
--- /dev/null
+++ b/configurations/system1_spyre.json
@@ -0,0 +1,66 @@
+{
+    "Exposes": [
+        {
+            "Address": "0x76",
+            "Bus": "$Bus",
+            "ChannelNames": [
+                "ch0",
+                "ch1",
+                "ch2",
+                "ch3"
+            ],
+            "Name": "SPYRE_MUX_$Bus - 72",
+            "Type": "PCA9546Mux"
+        },
+        {
+            "Address": "0x60",
+            "MuxChannel": {
+                "ChannelName": "ch1",
+                "MuxName": "SPYRE_MUX_$Bus - 72"
+            },
+            "Name": "SPYRE$Bus - 72 PVDD",
+            "Name1": "SPYRE$Bus - 72 PVCS",
+            "Type": "RAA228234"
+        },
+        {
+            "Address": "0x24",
+            "MuxChannel": {
+                "ChannelName": "ch3",
+                "MuxName": "SPYRE_MUX_$Bus - 72"
+            },
+            "PowerState": "On",
+            "Name": "SPYRE$Bus - 72 Ambient",
+            "Name1": "SPYRE$Bus - 72 Die",
+            "Type": "SPYRE"
+        },
+        {
+            "Address": "0x4B",
+            "MuxChannel": {
+                "ChannelName": "ch0",
+                "MuxName": "SPYRE_MUX_$Bus - 72"
+            },
+            "Name": "SPYRE$Bus - 72 TEMP1",
+            "Type": "TMP112"
+        },
+        {
+            "Address": "0x48",
+            "MuxChannel": {
+                "ChannelName": "ch1",
+                "MuxName": "SPYRE_MUX_$Bus - 72"
+            },
+            "Name": "SPYRE$Bus - 72 TEMP2",
+            "Type": "TMP112"
+        }
+    ],
+    "Name": "IBM System1 SPYRE $Bus - 72",
+    "Probe": "xyz.openbmc_project.Inventory.Decorator.Asset({'Model': '2F1E', 'Manufacturer': 'IBM'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "Manufacturer": "$Manufacturer",
+        "Model": "$Model",
+        "PartNumber": "$PartNumber",
+        "SerialNumber": "$SerialNumber",
+        "SparePartNumber": "$SparePartNumber"
+    },
+    "xyz.openbmc_project.Inventory.Item.System": {}
+}
-- 
2.43.0

